front-end/src/components/Register.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import axios from 'axios';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(8),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main,
+  },
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(1),
+  },
+  buttons: {
+    // display: 'block',//TODO: center buttons?
+  },
+}));
+
+const Register = () => {
+  const classes = useStyles();
+  const [registerSuccess, setregisterSuccess] = useState(false);
+  const [usernameInUse, setusernameInUse] = useState(false);
+  const [serverError, setserverError] = useState(false);
+
+  const handleRegister = (e) => {
+    e.preventDefault();
+    const { username, password } = e.target.elements;
+    console.log({ username: username.value, password: password.value });
+    const payload = {
+      username: username.value,
+      password: password.value,
+    };
+
+    const config = {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    };
+
+    axios
+      .post(`${process.env.REACT_APP_BACKEND_SERVICE}/register`, payload, config)
+      .then((res) => {
+        if (res.status === 200) {
+          console.log('SUCCESS');
+          setregisterSuccess(true);
+          setusernameInUse(false);
+          setserverError(false);
```

Comment:
since they're defaulted to false we can get rid of these

Reply:
So that's what I had originally. But the reason I put this in is because if you test by registering an existing user then registering a valid user, it will say: 

You successfully registered!
This username is already in use.

That is, those bools will stay `true` after the re-render so I had to toggle them back off, if that makes sense

Reply:
Ah....I forgot the hook persists the state throughout the app.